crates/cli/src/subcommands/generate/csharp.rs

```diff
@@ -430,69 +431,152 @@ fn autogen_csharp_access_funcs_for_struct(
             output,
             "public {csharp_field_name_pascal}UniqueIndex {csharp_field_name_pascal} = new();"
         );
-        writeln!(output);
     }
 
-    for idx in &schema.indexes {
-        match &idx.index_algorithm {
-            IndexAlgorithm::BTree(BTreeAlgorithm { columns }) => {
-                let col_pos = columns.head().unwrap().idx();
-                if constraints[&ColList::new(col_pos.into())].has_unique() {
-                    continue;
+    for index in btrees {
+        let fields: Vec<BTreeColumn<'_>> = index.columns
+            .iter()
+            .map(|col| {
+                let field = &product_type.elements[col.idx()];
+                let field_type = &field.algebraic_type;
+                match csharp_field_type(field_type) {
+                    None => todo!(),
+                    Some(x) => {
+                        let field_name = field.name.as_ref().expect("autogen'd tuples should have field names");
+                        let csharp_field_name_pascal = field_name.replace("r#", "").to_case(Case::Pascal);
+                        let range = match field_type {
+                            AlgebraicType::Product(pt) =>
+                                if pt.elements[0].name == Some("__address_bytes".into()) { ("Address.MinValue", "Address.MaxValue") }
+                                else if pt.elements[0].name == Some("__identity_bytes".into()) { ("Identity.MinValue", "Identity.MaxValue") }
+                                else { todo!() },
+                            AlgebraicType::String => ("\"\"", "\"\\uFFFF\\uFFFF\""),
+                            AlgebraicType::Bool => ("false", "true"),
+                            AlgebraicType::I8 => ("sbyte.MinValue", "sbyte.MaxValue"),
+                            AlgebraicType::U8 => ("byte.MinValue", "byte.MaxValue"),
+                            AlgebraicType::I16 => ("short.MinValue", "short.MaxValue"),
+                            AlgebraicType::U16 => ("ushort.MinValue", "ushort.MaxValue"),
+                            AlgebraicType::I32 => ("int.MinValue", "int.MaxValue"),
+                            AlgebraicType::U32 => ("uint.MinValue", "uint.MaxValue"),
+                            AlgebraicType::I64 => ("long.MinValue", "long.MaxValue"),
+                            AlgebraicType::U64 => ("ulong.MinValue", "ulong.MaxValue"),
+                            AlgebraicType::I128 => ("I128.MinValue", "I128.MaxValue"),
+                            AlgebraicType::U128 => ("U128.MinValue", "U128.MaxValue"),
+                            AlgebraicType::I256 => ("I256.MinValue", "I256.MaxValue"),
+                            AlgebraicType::U256 => ("U256.MinValue", "U256.MaxValue"),
+                            AlgebraicType::F32 => ("float.MinValue", "float.MaxValue"),
+                            AlgebraicType::F64 => ("double.MinValue", "double.MaxValue"),
```

Comment:
You should be able to use `is_integer() || is_float()` + `scalar_or_string_name` instead of repeating those.